assets/src/web-stories-block/block/api/useStoryApi.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo, useReducer } from 'react';
+import queryString from 'query-string';
+
+/**
+ * Internal dependencies
+ */
+import {
+  STORY_STATUSES,
+  STORY_SORT_OPTIONS,
+  ORDER_BY_SORT,
+  STORIES_PER_REQUEST,
+  STORY_STATUS,
+} from '../../../dashboard/constants';
+import storyReducer, {
+  defaultStoriesState,
+  ACTION_TYPES as STORY_ACTION_TYPES,
+} from '../../../dashboard/app/reducer/stories';
+import { ERRORS } from '../../../dashboard/app/textContent';
+
+const useStoryApi = (dataAdapter, { editStoryURL, storyApi }) => {
```

Comment:
Why is `editStoryURL` needed here?

Reply:
`editStoryURL` is passed down to the `storyReducer` which returns the `state` with two data points

```
      bottomTargetAction: `${editStoryURL}&post=${id}`,
      editStoryLink: `${editStoryURL}&post=${id}`, 
```
Though these are not required for our case, but it was returning urls with `undefined` parts. hence added it.